source/geod24/LocalRest.d

```diff
@@ -772,8 +818,398 @@ public final class RemoteAPI (API) : API
                     static if (!is(ReturnType!(ovrld) == void))
                         return res.data.deserializeJson!(typeof(return));
                 }
-                });
+            });
+        }
+}
+
+private
+{
+    bool hasLocalAliasing(Types...)()
+    {
+        import std.typecons : Rebindable;
+
+        // Works around "statement is not reachable"
+        bool doesIt = false;
+        static foreach (T; Types)
+        {
+            static if (is(T == Transceiver))
+            { /* Allowed */ }
+            else static if (is(T : Rebindable!R, R))
+                doesIt |= hasLocalAliasing!R;
+            else static if (is(T == struct))
+                doesIt |= hasLocalAliasing!(typeof(T.tupleof));
+            else
+                doesIt |= std.traits.hasUnsharedAliasing!(T);
+        }
+        return doesIt;
+    }
+
+    private template isSpawnable(F, T...)
+    {
+        template isParamsImplicitlyConvertible(F1, F2, int i = 0)
+        {
+            alias param1 = Parameters!F1;
+            alias param2 = Parameters!F2;
+            static if (param1.length != param2.length)
+                enum isParamsImplicitlyConvertible = false;
+            else static if (param1.length == i)
+                enum isParamsImplicitlyConvertible = true;
+            else static if (isImplicitlyConvertible!(param2[i], param1[i]))
+                enum isParamsImplicitlyConvertible = isParamsImplicitlyConvertible!(F1,
+                        F2, i + 1);
+            else
+                enum isParamsImplicitlyConvertible = false;
+        }
+
+        enum isSpawnable = isCallable!F && is(ReturnType!F == void)
+                && isParamsImplicitlyConvertible!(F, void function(LocalTransceiver, T))
+                && (isFunctionPointer!F || !hasUnsharedAliasing!F);
+    }
+
+    private LocalTransceiver spawnThread(F, T...)(F fn, T args)
+    if (isSpawnable!(F, T))
+    {
+        static assert(!hasLocalAliasing!(T), "Aliases to mutable thread-local data not allowed.");
+
+        auto thread_scheduler = new ThreadScheduler();
+        auto transceiver = new LocalTransceiver();
+        thread_scheduler.spawn({
+            thisLocalTransceiver = transceiver;
+            fn(transceiver, args);
+        });
+
+        return transceiver;
+    }
+}
+
+
+/*******************************************************************************
+
+    Transceiver device required for message exchange between threads.
+    Send and receive data requests, responses, commands, etc.
+
+*******************************************************************************/
+
+private class LocalTransceiver : Transceiver
```

Comment:
In essence, I think this class should not exist. It is just a thin wrapper around a `Channel` and doesn't add anything on its own. Instead `Channel` should be amended to comply with the interface you have in mind. I'll provide a longer explanation in the main thread.